lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -378,19 +459,90 @@ instance ToText FollowLog where
                 , "  Wallet/node slots: ", slot (NE.last hdrs)
                 , "/", slot tipHdr
                 ]
-        MsgWillRollback sl ->
-            "Will rollback to " <> pretty sl
         MsgWillIgnoreRollback sl reason ->
             "Will ignore rollback to " <> pretty sl
                 <> " because of " <> pretty reason
-
-instance HasPrivacyAnnotation FollowLog
-instance HasSeverityAnnotation FollowLog where
+        MsgWillRollback sl ->
+            "Will rollback to " <> pretty sl
+        MsgDidRollback requested actual -> mconcat
+            [ "Did rollback to "
+            , pretty actual
+            , " after request to rollback to "
+            , pretty requested
+            ]
+        MsgFailedRollingBack reason -> "Failed rolling back: " <>
+            reason
+        MsgFollowLog msg -> toText msg
+
+instance HasPrivacyAnnotation (FollowLog tr)
+instance HasSeverityAnnotation tr => HasSeverityAnnotation (FollowLog tr) where
     getSeverityAnnotation = \case
+        MsgStartFollowing -> Info
+        MsgHaltMonitoring -> Info
+        MsgFollowStats _ _ -> Info
         MsgFollowAction (ExitWith _) -> Error
         MsgFollowAction _ -> Debug
         MsgUnhandledException _ -> Error
         MsgSynced -> Debug
-        MsgApplyBlocks _ _ -> Info
+        MsgApplyBlocks _ _ -> Debug
+        MsgFollowLog msg -> getSeverityAnnotation msg
         MsgWillRollback _ -> Debug
+        MsgDidRollback _ _ -> Debug
+        MsgFailedRollingBack _ -> Error
         MsgWillIgnoreRollback _ _ -> Debug
+
+--
+-- Log aggregation
+-- Perhaps this could be done using iohk-monitoring as well.
+
+data RestorationStats = RestorationStats
+    { blocksApplied :: Int
+    , rollbacks :: Int
+    , t0 :: UTCTime
+    } deriving (Generic, Show, Eq)
+
+emptyStats :: UTCTime -> RestorationStats
+emptyStats = RestorationStats 0 0
+
+updateStats :: FollowLog tr -> RestorationStats -> RestorationStats
+updateStats msg s = case msg of
+    MsgApplyBlocks _tip blocks ->
+        s { blocksApplied = NE.length blocks + blocksApplied s}
+    MsgDidRollback _ _ ->
+        s { rollbacks = 1 + rollbacks s }
+    _ -> s
+
+-- TODO: What is actually interesting?
+--
+-- If we are not restored:
+-- - How fast we are restoring
+-- - Where we are on-chain
+-- - Where the node is
+-- - Where the current time is
+-- - How many rollbacks we encounter
+-- - What the worker itself is finding on-chain
+-- - How long the worker operations are
+-- - When we are restored
+--
+-- If we are restored:
+-- - If the node goes out of sync with the network
+-- - If the wallet goes out of sync with the node
+monitorStatus :: Tracer IO (FollowLog tr) -> IO (Tracer IO (FollowLog tr))
+monitorStatus tr = do
```

Comment:
> It may work better if log interpretation and transformation is done at the app top-level, all together, rather than inside each component. I'm not sure yet though.

Yeah, I was wondering about that too. Partly why I wondered if iohk-monitoring was the "right way", as I think their backends and configuration setup served exactly as that (top-level, all together transformation).

I suspect we can get by for now doing the transformation here.

Reply:
> For this ticket though, we just want to log interesting events and state changes.

IMO, being able to tell 
1. If the wallet is restoring slowly
2. If the wallet is rolling back in a loop (I _think_ the stats would be able to tell this)
is an important thing we lose when we remove the "on-each-batch-logs"